packages/jit-html/src/template-compiler.ts

```diff
@@ -146,6 +169,110 @@ export class TemplateCompiler implements ITemplateCompiler {
+      }
+      root = (template as HTMLTemplateElement).content;
+    } else {
+      root = template;
+    }
+    const localTemplates = toArray(root.querySelectorAll('template[as-custom-element]')) as HTMLTemplateElement[];
+    const numLocalTemplates = localTemplates.length;
+    if (numLocalTemplates === 0) { return; }
+    if (numLocalTemplates === root.childElementCount) {
+      throw new Error('The custom element does not have any content other than local template(s).');
+    }
+    const localTemplateNames: Set<string> = new Set();
+
+    function getTemplateName(localTemplate: HTMLTemplateElement): string {
+      const name = localTemplate.getAttribute(localTemplateIdentifier);
+      if (name === null || name === '') {
+        throw new Error('The value of "as-custom-element" attribute cannot be empty for local template'); /* TODO: use reporter/logger */
+      }
+      if (localTemplateNames.has(name)) {
+        throw new Error(`Duplicate definition of the local template named ${name}`); /* TODO: use reporter/logger */
+      } else {
+        localTemplateNames.add(name);
+      }
+      return name;
+    }
+    function getBindingMode(bindable: Element): BindingMode {
+      switch (bindable.getAttribute(LocalTemplateBindableAttributes.mode)) {
+        case 'oneTime':
+          return BindingMode.oneTime;
+        case 'toView':
+          return BindingMode.toView;
+        case 'fromView':
+          return BindingMode.fromView;
+        case 'twoWay':
+          return BindingMode.twoWay;
+        case 'default':
+        default:
+          return BindingMode.default;
+      }
+    }
+
+    for (const localTemplate of localTemplates) {
+      if (localTemplate.parentNode !== root) {
+        throw new Error('Local templates needs to be defined directly under root.'); /* TODO: use reporter */
+      }
+      const name = getTemplateName(localTemplate);
+
+      // eslint-disable-next-line @typescript-eslint/class-name-casing
+      const localTemplateType = class { };
+      const content = localTemplate.content;
+      const bindableEls = toArray(content.querySelectorAll('bindable'));
+      const bindableInstructions = Bindable.for(localTemplateType);
+      const properties: Set<string> = new Set();
+      const attributes: Set<string> = new Set();
+      for (const bindableEl of bindableEls) {
+        if (bindableEl.parentNode !== content) {
+          throw new Error('Bindable properties of local templates needs to be defined directly under root.'); /* TODO: use reporter */
+        }
+        const property = bindableEl.getAttribute(LocalTemplateBindableAttributes.property);
+        if (property === null) { throw new Error(`The attribute 'property' is missing in ${bindableEl.outerHTML}`); /* TODO: use reporter */ }
+        const attribute = bindableEl.getAttribute(LocalTemplateBindableAttributes.attribute) ?? void 0;
+        if (attributes.has(attribute!) || properties.has(property)) {
```

Comment:
```suggestion
        if ((attribute !== null && attributes.has(attribute)) || properties.has(property)) {
```

And just keep `attribute` `null` instead of `?? void 0`?

Reply:
Done